src/cpp/src/model_runner.hpp

```diff
@@ -114,13 +114,23 @@ class ModelRunner {
         subsequence_begins_data[0] = 0;
         block_indices_begins_data[0] = 0;
 
+        bool matmul_gathering_is_required = false;
+        int64_t gathering_current_index = 0;
+        std::vector<int64_t> gather_indice_values;
+        try {
+            std::ignore = m_request.get_tensor("gather_indices");
```

Comment:
alternatively, you can take `request.get_compiled_model().inputs()` and search input with name